app/models/content_item.rb

```diff
@@ -2,19 +2,40 @@ class ContentItem
   DEFAULT = ''.freeze
   attr_reader :path
 
+  class RedirectDetected < StandardError
+  end
+
   def initialize(path)
     @path = path
   end
 
   def title
     @title ||= begin
-      Services.content_store.content_item(@path).to_h['title'] || DEFAULT
+      content_store_data['title'] || DEFAULT
     rescue GdsApi::HTTPNotFound
       DEFAULT
     end
   end
 
+  def content_id
+    content_store_data.to_h['content_id']
+  end
+
   def url
     @url ||= PublicUrlService.absolute_url(path: @path)
   end
+
+  def content_store_data
+    @content_store_data ||= begin
+      response = Services.content_store.content_item(@path).to_h
+
+      unless response['base_path'] == @path
```

Comment:
could add a spec for this case

Reply:
Sure, I've added one here 25932d7a0b1c4203a2d1b0ecba5d643b5656b032 . Also, despite seeing this work yesterday, I actually had to tweak the implementation a bit (adding `.new`)...